Description: Fix issues when running the tests on the help for the manpage topic
 and when reading help in an environment installed via distro package
Author: Joseph Herlant <herlantj@gmail.com>
Forwarded: https://github.com/asciidoctor/asciidoctor/pull/2653
Last-Update: 2018-04-08

--- a/lib/asciidoctor/cli/options.rb
+++ b/lib/asciidoctor/cli/options.rb
@@ -134,6 +134,13 @@
             if topic == 'manpage'
               if ::File.exist?(manpage_path = (::File.join ::Asciidoctor::ROOT_PATH, 'man', 'asciidoctor.1'))
                 $stdout.puts(::IO.read manpage_path)
+              elsif ::File.exist?(manpage_path = (::File.join ::Asciidoctor::ROOT_PATH, '..', '..', 'share', 'man', 'man1', 'asciidoctor.1'))
+                $stdout.puts(::IO.read manpage_path)
+              elsif ::File.exist?(manpage_path = '/usr/share/man/man1/asciidoctor.1.gz')
+                require 'zlib'
+                gz = Zlib::GzipReader.open(manpage_path)
+                $stdout.puts(gz.read)
+                gz.close
               else
                 $stderr.puts 'asciidoctor: FAILED: man page not found; try `man asciidoctor`'
                 return 1
